google/cloud/aiplatform/_featurestores/featurestore.py

```diff
@@ -0,0 +1,527 @@
+
+    def __init__(
+        self,
+        featurestore_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing managed featurestore given a featurestore resource name or a featurestore ID.
+
+        Example Usage:
+
+            my_featurestore = aiplatform.Featurestore(
+                featurestore_name='projects/123/locations/us-central1/featurestores/my_featurestore_id'
+            )
+            or
+            my_featurestore = aiplatform.Featurestore(
+                featurestore_name='my_featurestore_id'
+            )
+
+        Args:
+            featurestore_name (str):
+                Required. A fully-qualified featurestore resource name or a featurestore ID.
+                Example: "projects/123/locations/us-central1/featurestores/my_featurestore_id"
+                or "my_featurestore_id" when project and location are initialized or passed.
+            project (str):
+                Optional. Project to retrieve featurestore from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve featurestore from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to retrieve this Featurestore. Overrides
+                credentials set in aiplatform.init.
+
+        Raises:
+            ValueError if the provided featurestore_name is not in form of a fully-qualified
+            featurestore resource name nor a featurestore ID.
+        """
+        if not (
+            featurestore_utils.validate_featurestore_name(featurestore_name)
+            or featurestore_utils.validate_id(featurestore_name)
+        ):
+            raise ValueError(
+                f"{featurestore_name} is not in form of a fully-qualified "
+                f"featurestore resource name nor a featurestore ID."
+            )
+
+        super().__init__(
+            project=project,
+            location=location,
+            credentials=credentials,
+            resource_name=featurestore_name,
+        )
+        self._gca_resource = self._get_gca_resource(resource_name=featurestore_name)
+
+    def get_entity_type(self, entity_type_id: str) -> "_featurestores.EntityType":
+        """Retrieves an existing managed entityType in this Featurestore.
+
+        Args:
+            entity_type_id (str):
+                Required. The managed entityType resource ID in this Featurestore.
+        Returns:
+            featurestores.EntityType - The managed entityType resource object.
+
+        Raises:
+            ValueError if the provided entity_type_id is not in form of an entity_type ID.
+        """
+        if not featurestore_utils.validate_id(entity_type_id):
```

Comment:
The `featurestore.get_entity_type` requires an `entity_type_id` only that's why there is an explicit `featurestore_utils.validate_id(entity_type_id):` in use. For the `EntityType` constructor, since it requires an `entity_type_name` and an optional `featurestore_id`, the validation is inside the `featurestore_utils.validate_and_get_entity_type_resource_ids`.